### ISSUE TITLE
question about training

### ISSUE THREAD

**Issue by 17864108656** [jakemcnaughton/CT-to-Synthetic-MRI-Generation#2]
Hello, during training, the validation set experienced fluctuations. I noticed that the CT images were not normalized. May I ask if there is any code related to data preprocessing?

**Comment by jakemcnaughton** [comment]
Thank you for the question. CT scans are in Hounsfield Units which are objective (not just a relative scale) and have a physical meaning. Due to this normalisation of the CT scans was not performed as this could cause important data loss. However, MRIs use a relative scale and as they are the target space, normalisation here is more important for the loss calculation. Therefore we normalised the MRI between [0,255].

The preprocessing pipeline used is detailed in section 2.1 of our paper that this code accompanies (https://www.mdpi.com/2673-7426/3/3/50). Here are the details of the pipeline used, links to the relevant packages and software are contained in the paper:

1. Registration of the CT to the MRI using FSL Flirt with 12 degrees of freedom and a mutual information cost function.
2. Registration of the MRI to the ICBM152 T1w linear atlas using FSL Flirt with 12 degrees of freedom and a correlation ratio cost function, and saving the transformation matrix.
3. Applying the transformation matrix from step 2 to the resultant CT from step 1.
4. Apply the runhdbet function of HD-BET to the resultant MRI from step 2. The resultant extraction brain and brain mask are then saved.
5. Using pixelwise multiplication between the resultant CT scan from step 3 and the brain mask from step 4 to extract the brain from the CT.

After the data were processed through the pipeline, the data underwent manual quality control by overlaying the CT and MRI of each patient and visually inspecting the alignment over multiple slices. Data that could not be successfully registered from CT to MRI in FSL Flirt were attempted to be registered in 3D Slicer. To ensure consistency between the images, all MRI images were set to a data range of [0, 255] by dividing the intensities of each image by its maximum intensity and multiplying by 255. CT scans were kept in their original intensity range as they were in Hounsfield Units.
